packages/core/CalendarManager.ts

```diff
@@ -73,11 +67,8 @@ export const getConnectedCalendars = async (
           })
           .sortBy(["primary"])
           .value();
-
-        if (primary && destinationCalendar) {
-          destinationCalendar.primaryEmail = primary.email;
-          destinationCalendar.integrationTitle = integration.title;
-        }
+        const primary!: IntegrationCalendar =
+          calendars.find((item) => item.primary) ?? calendars.find((cal) => cal !== undefined);
```

Comment:
Added this as otherwise the fallback was missing causing the errors suggested in the related issue